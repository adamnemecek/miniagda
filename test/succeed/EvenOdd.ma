-- 2010-08-28 mutual data types

mutual {

  data Even : Set
  { ev0 : Even
  ; evS : Odd -> Even
  }

  data Odd : Set
  { oddS : Even -> Odd
  }

}

data Nat : Set
{ zero : Nat
; suc  : Nat -> Nat
}

mutual {

  fun evenToNat : Even -> Nat
  { evenToNat ev0 = zero
  ; evenToNat (evS o) = suc (oddToNat o)
  }

  fun oddToNat : Odd -> Nat
  { oddToNat (oddS e) = suc (evenToNat e)
  }
}
