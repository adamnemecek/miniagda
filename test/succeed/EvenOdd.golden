MiniAgda by Andreas Abel and Karl Mehltretter
--- opening "EvenOdd.ma" ---
--- scope checking ---
--- type checking ---
type  Even : Set
term  Even.ev0 : < Even.ev0 : Even >
term  Even.evS : ^(y0 : Odd) -> < Even.evS y0 : Even >
type  Odd : Set
term  Odd.oddS : ^(y0 : Even) -> < Odd.oddS y0 : Odd >
type  Nat : Set
term  Nat.zero : < Nat.zero : Nat >
term  Nat.suc : ^(y0 : Nat) -> < Nat.suc y0 : Nat >
term  evenToNat : Even -> Nat
term  oddToNat : Odd -> Nat
{ evenToNat Even.ev0 = Nat.zero
; evenToNat (Even.evS o) = Nat.suc (oddToNat o)
}
{ oddToNat (Odd.oddS e) = Nat.suc (evenToNat e)
}
--- evaluating ---
--- closing "EvenOdd.ma" ---
